app/labor/rate_limit_checker.rb

```diff
@@ -26,6 +26,12 @@ def track_image_uploads
     Rails.cache.write("#{@user.id}_image_upload", count, expires_in: 30.seconds)
   end
 
+  def track_article_updates
+    count = Rails.cache.read("#{@user.id}_article_update").to_i
+    count += 1
+    Rails.cache.write("#{@user.id}_article_update", count, expires_in: 1.day)
```

Comment:
To eliminate the multiple lines you could use [increment here](https://api.rubyonrails.org/classes/ActiveSupport/Cache/Store.html#method-i-increment)
```
Rails.cache.increment("#{@user.id}_article_update", expires_in: 1.day)
```

Reply:
+1 on this, it will also be atomic as, between read and write another thread could read the same value or write the same value twice. See https://redis.io/commands/incr

Reply:
(pps. `track_image_uploads` could use the same change :D)